mava/networks/utils/sable/discrete_trainer_executor.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Optional, Tuple
+
+import chex
+import distrax
+import jax
+import jax.numpy as jnp
+from flax import linen as nn
+
+
+def train_encoder_parallel(
+    encoder: nn.Module,
+    obs: chex.Array,
+    hstate: chex.Array,
+    dones: chex.Array,
+    timestep_id: chex.Array,
+) -> Tuple[chex.Array, chex.Array, chex.Array]:
+    """Parallel encoding for discrete action spaces."""
+    # Apply the encoder
+    v_loc, obs_rep, updated_hstate = encoder(obs, hstate, dones, timestep_id)
+    return v_loc, obs_rep, updated_hstate
```

Comment:
the issue we want to unfiy the call of functions, for example if this one do not exist it will cause the use of an if statement in the sable network because the input is different from train_encoder_chunkwise